meteor/client/ui/Shelf/DashboardPieceButton.tsx

```diff
@@ -100,9 +102,47 @@ export class DashboardPieceButtonBase<T = {}> extends MeteorReactComponent<
 
 	getThumbnailUrl = (): string | undefined => {
 		const { piece } = this.props
-		if (this.props.mediaPreviewUrl && piece.contentMetaData) {
-			if (piece.contentMetaData && piece.contentMetaData.previewPath && this.props.mediaPreviewUrl) {
-				return this.props.mediaPreviewUrl + 'media/thumbnail/' + encodeURIComponent(piece.contentMetaData.mediaId)
+		if (piece.expectedPackages) {
+			// use Expected packages:
+			// Just use the first one we find.
+			// TODO: support multiple expected packages?
+			let packagePath: string | undefined
+			let thumbnailContainerId: string | undefined
+			for (const expectedPackage of piece.expectedPackages) {
+				if (expectedPackage.type === ExpectedPackage.PackageType.MEDIA_FILE) {
+					packagePath = expectedPackage.content.filePath
+					thumbnailContainerId = expectedPackage.sideEffect.thumbnailContainerId
+				} else if (expectedPackage.type === ExpectedPackage.PackageType.QUANTEL_CLIP) {
+					packagePath = expectedPackage.content.guid || expectedPackage.content.title
+					thumbnailContainerId = expectedPackage.sideEffect.thumbnailContainerId
+					break
+				}
+			}
+			if (packagePath && thumbnailContainerId) {
+				const packageContainer = this.props.studio?.packageContainers[thumbnailContainerId]
+				if (packageContainer) {
+					// Look up an accessor we can use:
+					for (const accessor of Object.values(packageContainer.container.accessors)) {
+						if (accessor.type === Accessor.AccessType.HTTP && accessor.baseUrl) {
+							// TODO: add fiter for accessor.networkId ?
+							return [
+								accessor.baseUrl.replace(/\/$/, ''), // trim trailing slash
+								encodeURIComponent(
+									packagePath
+										.replace(/^\//, '') // trim leading slash
+										.replace(/(\.[^.]+$)/, '.png') // replace file extension with png
```

Comment:
Like with previews, we shouldn't assume PNG files, we shouldn't assume there is a file extension and string mangling for producing an encodeURIComponent-string has a bad smell for me.

Reply:
Addressed in https://github.com/nrkno/tv-automation-server-core/pull/408/commits/cb3a600fdd0ab9f728c5bd6c1640f8bd3002b0ac